components/org.wso2.carbon.security.caas/src/main/java/org/wso2/carbon/security/caas/user/core/store/CredentialStoreImpl.java

```diff
@@ -127,14 +120,17 @@ public AuthenticationContext authenticate(Callback[] callbacks) throws Authentic
         }
 
         // TODO: Resolve domain
-        Map<String, CredentialStoreConnector> credentialStoreConnectorsMap =
-                null;
+        Map<String, CredentialStoreConnector> credentialStoreConnectorsMap;
+
         try {
-            credentialStoreConnectorsMap = this.domainManager.getDefaultDomain().getCredentialStoreConnectorMap();
-        } catch (DomainManagerException e) {
+            credentialStoreConnectorsMap = resolveDomain(callbacks).getCredentialStoreConnectorMap();
+        } catch (CredentialStoreException e) {
             credentialStoreConnectorsMap = Collections.emptyMap();
+            log.error(e.getMessage());
         }
 
+        AuthenticationFailure authenticationFailure = new AuthenticationFailure("Invalid user credentials.");
+
```

Comment:
This is required since the CredentialStoreExceptions are suppressed to the AuthenticationFailure and is thrown after the loop finished